### ISSUE TITLE
About domain-specific and domain-invariant block

### ISSUE THREAD

**Issue by DawnJYe** [gist-ailab/domain-specific-block-selection-and-paired-view-pseudo-labeling-for-online-TTA#2]
Hello！This paper mentions that adding Gaussian noise represents domain shift, and high similarity of category prototypes implies that the features extracted before and after adding noise are similar across different categories, overcoming domain shift and extracting domain-independent semantic features. However, the text points out that high similarity indicates involvement of domain-specific feature extraction. Am I misunderstanding this? Please clarify my doubt, thank you!

**Comment by birdomi** [comment]
Thank you for your attention in our work.

Our paper is motivated by the insight that domain-invariant features should be maintained even after the test-time adaptation.
Thus, we only select the block that does not used to domain-invariant feature (i.e., semantic feature) for entropy minimization.

So, we would say "Gaussian noise represents domain shift, and high similarity of category prototypes implies that the features extracted before and after adding noise are similar across different categories, overcoming domain shift and extracting domain-independent semantic features." as you mentioned it. 

Also, "we select the blocks that maintain high similarity, indicating that the blocks do not involve in domain-invariant
feature extraction (i.e., involves in domain-specific feature extraction)." in our paper (Section 3.1) is wrongly written (should be written as domain-invariant -> domain-specific and vice versa). We will fix this sentence as in method section, which is correctly written.

We are sorry for our mistake.

**Comment by birdomi** [comment]
We correct the arxiv paper.
Sorry again for our mistake.

**Comment by DawnJYe** [comment]
Thanks for your reply！Here remains a point that puzzles me, as pointed out in Section 3.1 of your paper: "the selected blocks are updated by minimizing entropy for the given test data to adjust the domain-specific feature extraction to the current corruption" and " we can adjust domain-specific feature extraction without disrupting domain-invariant feature extraction unlike previous method".Here, the operation should be performed on the domain-specific feature block to obtain domain-invariant features. However, you previously mentioned that the selected block is one with high similarity, namely the domain-invariant feature extraction block. So, in the end, is the selected block domain-specific or domain-invariant? According to my understanding, shouldn't we select blocks with low similarity, that is, domain-specific feature extraction blocks, and then perform operations on them?

**Comment by birdomi** [comment]
Oh, sorry again for my bad.

My first response was wrong. I must have been misread my sentence :( 

The sentence "we select the blocks that maintain high similarity, indicating that the blocks do not involve in domain-invariant
feature extraction (i.e., involves in domain-specific feature extraction)" is correct sentence.

We argue that similarity between two prototype vectors (one with original training images and other one with noise-added images) represent how the block involves in domain-invariant feature extraction. Thus, after the similarity for each block is calculated (i.e., degree of domain-invariant feature extraction), we select the block with high similarity. This selection is based on the assumption that the block involves in domain-invariant feature extraction would not involve in domain-specific feature extraction. Thus, updating the high-similarity block is helpful for adjusting domain difference.

In summary, the selected block is domain-specific and we update its parameter using entropy minimization.

Thank you for your attention in our work.

**Comment by Lzyin** [comment]
Thank you for your interesting work!

I also have a confusion. In section 3.2, it is mentioned that "_A high similarity indicates that the block is adapted on Gaussian noise added images without modifying domain-invariant feature space_." My understanding is that a high similarity indicates that the block extracts domain-invariant features, and therefore, the block's parameters are not affected by the training data with added Gaussian noise. So, blocks with high similarity should be the ones extracting domain-invariant features? However, this understanding seems to be the opposite of what is stated in the paper?

I sincerely hope to get your reply.

**Comment by birdomi** [comment]
Thank you for your attention in our work.

We select the block with high similarity based on the two following assumptions:
- First, the training image with added Gaussian noise and the original training image have the similar domain-invariant features (e.g., shape of the object does not changed by noise) but different domain-specific features. 
- Second, the similarity calculation is based on the features from the last block even when we update the block in shallow layer. So, we believe that high similarity indicates the similarity between domain-invariant features before and after the update of target block.

Based on the above assumptions, when we update the block that extracts domain-specific features using noise-added images, the update is stable and does not decrease the similarity of domain-invariant features before and after the update (i.e., high similarity). However, when we update the block that extracts domain-invariant features, the update is unstable and may decrease the similarity.

Hope this comment is helpful.

**Comment by Lzyin** [comment]
I get it. Thank you again for your reply.
